config/kraft/reconfig-server.properties

```diff
@@ -0,0 +1,132 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
```

Comment:
It's unclear why we're adding a new file

Reply:
@jsancio we should use the same approach here in `server.properties` as in `broker.properties` and `controller.properties`

As I commented above, the simplest approach is just to have a line for `controller.quorum.voters`, but have it be commented out, with an explanation about how to use the old style if desired. That would avoid the need for a separate file.

Reply:
> It's unclear why we're adding a new file

This file is reference in Dockerfile and a few other places. I haven't had a chance to fix all of those places to use kraft.version 0. I instead created this jira: https://issues.apache.org/jira/browse/KAFKA-17576

```bash
rg "kraft\/server.properties"
docker/native/Dockerfile
66:COPY --chown=appuser:root --from=build-native-image /app/kafka/config/kraft/server.properties /etc/kafka/docker/

docker/docker_official_images/3.7.0/jvm/Dockerfile
81:    cp /opt/kafka/config/kraft/server.properties /etc/kafka/docker/server.properties; \

docker/docker_official_images/3.7.0/jvm/jsa_launch
20:KAFKA_JVM_PERFORMANCE_OPTS="-XX:ArchiveClassesAtExit=storage.jsa" opt/kafka/bin/kafka-storage.sh format -t $KAFKA_CLUSTER_ID -c opt/kafka/config/kraft/server.properties
22:KAFKA_JVM_PERFORMANCE_OPTS="-XX:ArchiveClassesAtExit=kafka.jsa" opt/kafka/bin/kafka-server-start.sh opt/kafka/config/kraft/server.properties &

docker/jvm/Dockerfile
81:    cp /opt/kafka/config/kraft/server.properties /etc/kafka/docker/server.properties; \

docker/jvm/jsa_launch
20:KAFKA_JVM_PERFORMANCE_OPTS="-XX:ArchiveClassesAtExit=storage.jsa" opt/kafka/bin/kafka-storage.sh format -t $KAFKA_CLUSTER_ID -c opt/kafka/config/kraft/server.properties
22:KAFKA_JVM_PERFORMANCE_OPTS="-XX:ArchiveClassesAtExit=kafka.jsa" opt/kafka/bin/kafka-server-start.sh opt/kafka/config/kraft/server.properties &

docs/streams/quickstart.html
133:<pre><code class="language-bash">$ bin/kafka-storage.sh format -t $KAFKA_CLUSTER_ID -c config/kraft/server.properties</code></pre>
139:<pre><code class="language-bash">$ bin/kafka-server-start.sh config/kraft/server.properties</code></pre>

README.md
98:    ./bin/kafka-storage.sh format -t $KAFKA_CLUSTER_ID -c config/kraft/server.properties
99:    ./bin/kafka-server-start.sh config/kraft/server.properties
```

Reply:
> As I commented above, the simplest approach is just to have a line for controller.quorum.voters, but have it be commented out, with an explanation about how to use the old style if desired. That would avoid the need for a separate file.

See my other comment. I am not sure if we need to document all of the options. Kafka and KRaft support a lot of configurations that are not enumerated in `kraft/server.properties`.

Reply:
For now, I guess we can add a new file, although it's not ideal.